src/users/users.service.ts

```diff
@@ -261,4 +262,27 @@ export class UsersService {
         return paginate<Users>(queryBuilder, options);
     }
 
+    /**
+     * 
+     * @param empID ID of the logged in user
+     * @param newUser new User object to update old user
+     * @returns the new User object which was used to update the user
+     */
+    async editUserDetails(requester: Users, employeeId: number, newUser: Users){
+        if(requester.employeeId !== employeeId && requester.role !== Role.Admin){
+            throw new UnauthorizedException();
+        }
+        const user = await this.usersRepository.findOne({employeeId, companyId: requester.companyId});
+        user.firstName = newUser.firstName;
+        user.lastName = newUser.lastName;
+        if (requester.role === Role.Admin) {
+            user.positionTitle = newUser.positionTitle;
+            user.startDate = newUser.startDate;
+            user.isManager = newUser.isManager
+        }
+        return newUser;
```

Comment:
Are you saving the new user details to the database? Does typeorm do that automatically?

Reply:
uhh no I'm not apparently lol, must've not got pushed up. One sec

Reply:
should be good now, my bad